### PR TITLE
Add no_dupe_else_if rule

### PR DESCRIPTION
**Pull request by sladyn98**
Add [no_duple_else_if](https://eslint.org/docs/latest/rules/no-dupe-else-if)

Related issue https://github.com/Boshen/oxc/issues/123

### CODE REVIEW

**Comment by Boshen** [comment]
Thank you for the PR! I think you need to rename the file to `no_dupe_else_if.rs`. And then can start running `cargo test -p oxc_linter` and `cargo insta accept`.

**Comment by sladyn98** [comment]
@Boshen I am actually still getting a hang of the codebase so I would really appreciate if I could get some help here
```
error[E0599]: no method named `iter` found for reference `&oxc_ast::ast::Statement<'_>` in the current scope
  --> crates/oxc_linter/src/rules/no_dupe_else_if.rs:50:41
   |
50 |                 for stmt in else_branch.iter() {
   |                                         ^^^^ method not found in `&Statement<'_>`

error[E0599]: no method named `static_hash` found for enum `oxc_ast::ast::Expression` in the current scope
  --> crates/oxc_linter/src/rules/no_dupe_else_if.rs:52:54
   |
52 |                         let hash = else_if_stmt.test.static_hash();
   |                                                      ^^^^^^^^^^^ method not found in `Expression<'_>`


```
I could not really understand how to see if the else_branch is iterable because it seems its not, I tried looking at the other rules for some inspiration but was not able to find any

**Comment by Boshen** [comment]
@sladyn98 I don't quite understand your intention on iterating the `else_branch`, but here are a few pointers:
* you can still look at the eslint implementation to see how the algorithm works https://github.com/eslint/eslint/blob/main/lib/rules/no-dupe-else-if.js
* If you need to go up the AST, you need to use `ancestors`

https://github.com/Boshen/oxc/blob/7899cf545dd6f4b6f88c0c4a153b574dce7ad793/crates/oxc_linter/src/context.rs#L111-L113

A usecase:

https://github.com/Boshen/oxc/blob/c16f858ae83c8ef0feaa349b75a5d43f0b34a33a/crates/oxc_semantic/src/checker/javascript.rs#L205-L206

* If you need to look at the parent Node, use `parent_kind`

https://github.com/Boshen/oxc/blob/7899cf545dd6f4b6f88c0c4a153b574dce7ad793/crates/oxc_linter/src/context.rs#L101-L103

* For hashing, use `calculate_hash` from `crates/oxc_linter/src/ast_util.rs`

File: crates/oxc_linter/src/rules/no_dup_else_if.rs
Comment (Boshen) [inline crates/oxc_linter/src/rules/no_dup_else_if.rs:1]: The file name should be `no_dupe_else_if`, otherwise the macro won't pick it up.

**Comment by Boshen** [comment]
Closing due to inactivity, please feel free to rework on this 😁

**Pull request closed without merging**